lib/temporal/metadata.rb

```diff
@@ -6,34 +6,11 @@
 
 module Temporal
   module Metadata
-    ACTIVITY_TYPE = :activity
-    WORKFLOW_TASK_TYPE = :workflow_task
 
     class << self
       include Concerns::Payloads
 
-      def generate(type, data, namespace)
-        case type
-        when ACTIVITY_TYPE
-          activity_metadata_from(data, namespace)
-        when WORKFLOW_TASK_TYPE
-          workflow_task_metadata_from(data, namespace)
-        else
-          raise InternalError, 'Unsupported metadata type'
-        end
-      end
-
-      private
-
-      def headers(fields)
-        result = {}
-        fields.each do |field, payload|
-          result[field] = from_payload(payload)
-        end
-        result
-      end
-
-      def activity_metadata_from(task, namespace)
+      def generate_activity_metadata(task, namespace=nil)
```

Comment:
And here ^

Reply:
Verified all call sites specify a value and removed the default